benchmarks/synth-bm/src/account.rs

```diff
@@ -219,24 +258,16 @@ pub async fn create_sub_accounts(args: &CreateSubAccountsArgs) -> anyhow::Result
     info!("Querying nonces of newly created sub accounts.");
 
     // Nonces of new access keys are set by nearcore: https://github.com/near/nearcore/pull/4064
-    // Query them from the rpc to write `Accounts` with valid nonces to disk.
-    // TODO use `JoinSet`, e.g. by storing accounts in map instead of vec.
-    let mut get_access_key_tasks = Vec::with_capacity(sub_accounts.len());
-    // Use an interval to avoid overwhelming the node with requests.
-    let mut interval = time::interval(Duration::from_micros(150));
-    for account in sub_accounts.clone().into_iter() {
-        interval.tick().await;
-        let client = client.clone();
-        get_access_key_tasks.push(tokio::spawn(async move {
-            view_access_key(&client, account.id.clone(), account.public_key.clone()).await
-        }))
-    }
+    // Query them from the rpc to write `Accounts` with valid nonces to disk
+    sub_accounts = update_account_nonces(
+        client.clone(),
+        sub_accounts,
+        1_000_000 / args.interval_duration_micros,
```

Comment:
Any reason this code goes from duration to Hz and then back within the function?

Reply:
Might be a side effect of having multiple PRs in flight.